.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,19 @@
+name: unit tests
+on:
+  workflow_dispatch: # run CI when triggered manually
+  workflow_call: # run when called from another workflow
+defaults:
+  run:
+    shell: bash
+jobs:
+  test:
+    strategy:
```

Comment:
Nitpick: In this case where the matrix doesn't actually do anything since it only has one option configured, I would prefer removing it to not have unneeded LoC